buildpipe/pipeline.py

```diff
@@ -248,29 +246,31 @@ def check_for_trigger_steps(build_steps):
             del step['env']
 
 
-def compile_steps(config: box.Box) -> box.Box:
+def compile_steps(config: Dict) -> Dict:
     validate_config(config)
     branch = get_git_branch()
     projects = get_affected_projects(branch, config)
-    can_autodeploy = check_autodeploy(config.deploy.to_dict())
+    can_autodeploy = check_autodeploy(config.get('deploy', {}))
     scope_fn = dict(project=generate_project_steps, stair=generate_stair_steps)
 
     steps = []
-    previous_stair = box.Box({'continue_on_failure': False})
-    for stair in iter_stairs(config.stairs, can_autodeploy):
+    previous_stair = {'continue_on_failure': False}
+    for stair in iter_stairs(config.get('stairs', []), can_autodeploy):
         stair_projects = list(iter_stair_projects(stair, projects))
         if stair_projects:
             steps += generate_wait_step(previous_stair)
-            steps += generate_block_step(config.block.to_dict(), stair, stair_projects)
-            steps += scope_fn[stair.scope](stair, stair_projects)
+            steps += generate_block_step(config.get('block', {}), stair, stair_projects)
+            steps += scope_fn[stair['scope']](stair, stair_projects)
         check_for_trigger_steps(steps)
         previous_stair = stair
 
-    return box.Box({'steps': steps})
+    return {'steps': steps}
 
 
 def create_pipeline(infile: str, outfile: str, dry_run: bool = False) -> NoReturn:
-    config = box.Box.from_yaml(filename=infile, **BOX_CONFIG)
+    with open(infile, 'r') as in_f:
+        config = yaml.load(in_f, Loader=ruamel.yaml.Loader)
```

Comment:
`ruamel.yaml` looks like it has a different preferred interface starting version 0.15.
[This page](https://yaml.readthedocs.io/en/latest/api.html) documents the change.
The documentation is also a little bit flaky about what loaders exist.
I couldn't find any docs on what this loader is.

Reply:
Also, since the import `from ruamel import yaml` exists, this can be written as follows:
```suggestion
        config = yaml.load(in_f, Loader=yaml.Loader)
```